ee/clickhouse/queries/funnels/funnel_unordered.py

```diff
@@ -107,7 +107,8 @@ def _get_step_times(self, max_steps: int):
 
         for i in range(1, max_steps):
             conditions.append(
-                f"if(isNotNull(conversion_times[{i+1}]), dateDiff('second', conversion_times[{i}], conversion_times[{i+1}]), NULL) step_{i}_conversion_time"
+                f"if(isNotNull(conversion_times[{i+1}]) AND conversion_times[{i+1}] <= conversion_times[{i}] + INTERVAL {self._filter.funnel_window_interval} {self._filter.funnel_window_interval_unit_ch()}, "
+                f"dateDiff('second', conversion_times[{i}], conversion_times[{i+1}]), NULL) step_{i}_conversion_time"
```

Comment:
Q: Why are we interpolating conversion_times instead of passing them as params?

Reply:
`conversion_times` is an array generated in SQL. We are interpolating just the indexes, which determine the ith conversion time.

We need to do this because in the unordered funnel, we don't know apriori which event will be the ith event to happen.